es/packages/protoc-gen-es-cybozu-validate/src/field.ts

```diff
@@ -0,0 +1,607 @@
+import { DescField, DescOneof, ScalarType } from "@bufbuild/protobuf";
+import {
+  GeneratedFile,
+  ImportSymbol,
+  findCustomMessageOption,
+} from "@bufbuild/protoplugin/ecmascript";
+import { localName, makeJsDoc } from "@bufbuild/protoplugin/ecmascript";
+import {
+  DoubleRules,
+  Int32Rules,
+  Int64Rules,
+  Uint32Rules,
+  Uint64Rules,
+  FloatRules,
+  FieldRules,
+  BytesRules,
+  ItemsRules,
+  EnumRules,
+  StringRules,
+} from "@cybozu/protobuf/dist/validate/options_pb";
+import { capitalizeFirstLetter } from "./string-utils";
+
+type NumberRules =
+  | FloatRules
+  | DoubleRules
+  | Int32Rules
+  | Int64Rules
+  | Uint32Rules
+  | Uint64Rules;
+
+type Rules = NonNullable<FieldRules["type"]["value"]>;
+
+function renderFieldValidator(f: GeneratedFile, field: DescField) {
+  const customOption = findCustomMessageOption(
+    field,
+    1179,
+    // @ts-expect-error -- foo
+    FieldRules
+  );
+  // no available rules for bools
+  if (field.scalar === ScalarType.BOOL || !customOption) {
+    return;
+  }
+  const localFieldName = localName(field);
+  const capitalizedFieldName = capitalizeFirstLetter(localFieldName);
+
+  f.print(makeJsDoc(field, "  "));
+  f.print`  validate${capitalizedFieldName}(value) {`;
+
+  renderField(
+    f,
+    field,
+    // @ts-expect-error -- foo
+    customOption
+  );
+
+  f.print`  },`;
+}
+
+function renderField(
+  f: GeneratedFile,
+  field: DescField,
+  customOption: FieldRules | undefined
+) {
+  if (field.optional) {
+    f.print("    if (value == null) {");
+    f.print("      return;");
+    f.print("    }");
+  }
+
+  switch (field.fieldKind) {
+    case "scalar":
+      renderScalar(f, field, customOption);
+      break;
+    case "enum":
+      renderEnum(
+        f,
+        field,
+        customOption?.items.value,
+        customOption?.type.value as EnumRules
+      );
+      break;
+    case "map":
+      renderMap(
+        f,
+        field,
+        customOption?.items.value,
+        customOption?.type.value as FieldRules
+      );
+      break;
+    case "message":
+      renderMessageField(f, field, "value");
+      break;
+  }
+}
+
+function renderScalar(
```

Comment:
Can you separate this file into some small pieces for readability, such as `/field/index.ts`, `/field/scalar.ts` exporting `renderScalar()`, `/field/enum.ts` exporting `renderEnum()`, `/field/map` exporting `renderMap()`, and `/field/message.ts` exporting `renderMessageField()`?